lnclient/ldk/ldk.go

```diff
@@ -491,20 +491,19 @@ func (ls *LDKService) SendPaymentSync(ctx context.Context, invoice string, amoun
 	preimage := ""
 
 	timeout := time.Second * time.Duration(*timeoutSeconds)
-	ctx, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
+	timer := time.NewTimer(timeout)
```

Comment:
why add a timer here compared to using it directly in the case like it's done in other places?

Reply:
Because if we get any event on LDK which doesn't cause `SendPaymentSync` to return, it would go via the for loop again, checking 50s from the time it looped again, extending the actual timeout...

Reply:
Ah right, thanks!